theano/tensor/basic.py

```diff
@@ -5943,13 +5954,45 @@ def infer_shape(self, node, shapes):
 
 
 def diagonal(a, offset=0, axis1=0, axis2=1):
+    """
+    A helper function for Diagonal Op.
+
+    Parameters
+    ----------
+    a
+        A tensor variable with x.ndim >= 2.
+
+    offset
+        Indicates which diagonal to extract, "0" as main diagonal, positive as
+        diagonals in the upper triangle, and vice versa.
+
+    axis1, axis2
+        Indicates on which dimensions to extract the diagonals.
+
+    Returns
+    -------
+    vector
+        A vector representing the diagonal elements.
+    """
     if (offset, axis1, axis2) == (0, 0, 1):
         return theano.tensor.nlinalg.extract_diag(a)
     return Diagonal(offset, axis1, axis2)(a)
 
 
 class Diag(Op):
+    """
+    An op that transfers a vector into a matrix containing values in that
+    vector as its diagonal. It does the inverse of Diagonal.
 
+    Usage: T.diag()(x)
+
+    x should be a tensor vector. The parenthesis in the front should indicate
+    which main diagonal the vector value goes into. By default it is set to (
+    "0", which corresponds to setting the values of x to the main diagonal in
+    the returned matrix. Currently it only accepts "0", as other diagonals
+    hasn't been implemented for now.
```

Comment:
Do you intend to modify the Op? So far I don't see anything about axis diagonal set to 0 by default.